skia/recorder/test/src/archive_test.cpp

```diff
@@ -0,0 +1,61 @@
+#define private public // Expose private fields/methods.
+#include "catch.hpp"
+#include "archive.hpp"
+#include <cstdio> // std::remove()
+
+TEST_CASE("Read the correct number of bytes")
+{
+	auto bytes = Archive::read_file("./static/51x50.riv");
+	REQUIRE(bytes.size() == 76);
+}
+
+TEST_CASE("Non-finalized Archive doesnt create zip")
+{
+	auto file_bytes = Archive::read_file("./static/51x50.riv");
+	std::string archive_location("./static/archive_test.zip");
+	Archive* archive = new Archive(archive_location);
+	archive->add_buffer("buffer.riv", file_bytes);
+	delete archive;
+
+	zip* zip_ptr_dealloc = archive->zip_archive;
+	REQUIRE(zip_ptr_dealloc == nullptr);
+
+	std::ifstream infile(archive_location);
+	REQUIRE(!infile.good());
+}
+
+TEST_CASE("Empty Archive doesn't create a file")
+{
+	std::string archive_location("./static/archive_test.zip");
+	Archive archive = Archive(archive_location);
+
+	archive.finalize();
+	archive.~Archive();
```

Comment:
My eyes. Stop the bleeding. What is this? Help my brain stop bleeding.

Reply:
<img src="https://media.giphy.com/media/OHygvFftZxYYw/giphy.gif" />